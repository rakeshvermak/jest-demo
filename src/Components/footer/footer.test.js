import React from 'react';
import { render,cleanup } from '@testing-library/react';
import FooterPage from './footer';

afterEach(cleanup);

test('should renders Copyright', () => {
  const { getByText } = render(<FooterPage />);
  const linkElement = getByText(/Copyright:/i);
  expect(linkElement).toBeInTheDocument();
});

//Taking snapshots

it('should renders footer snapshot',()=>{
    const {asFragment} = render(<FooterPage />);
    expect(asFragment()).toMatchSnapshot();
})
