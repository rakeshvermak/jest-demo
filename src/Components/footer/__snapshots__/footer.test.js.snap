// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders footer snapshot 1`] = `
<DocumentFragment>
  <footer
    class="page-footer blue font-small pt-4 mt-4"
    data-test="footer"
  >
    <div
      class="container-fluid text-center text-md-left"
      data-test="container"
    >
      <div
        class="row"
        data-test="row"
      >
        <div
          class="col-md-6"
          data-test="col"
        >
          <h5
            class="title"
          >
            Footer Content
          </h5>
          <p>
            Here you can use rows and columns here to organize your footer content.
          </p>
        </div>
        <div
          class="col-md-6"
          data-test="col"
        >
          <h5
            class="title"
          >
            Links
          </h5>
          <ul>
            <li
              class="list-unstyled"
            >
              <a
                href="#!"
              >
                Link 1
              </a>
            </li>
            <li
              class="list-unstyled"
            >
              <a
                href="#!"
              >
                Link 2
              </a>
            </li>
            <li
              class="list-unstyled"
            >
              <a
                href="#!"
              >
                Link 3
              </a>
            </li>
            <li
              class="list-unstyled"
            >
              <a
                href="#!"
              >
                Link 4
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="footer-copyright text-center py-3"
    >
      <div
        class="container-fluid"
        data-test="container"
      >
        Â© 2020 Copyright: 
        <a
          href="https://www.shyamhub.com"
        >
           ShyamHub.com 
        </a>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
